delta-app/src/main/java/io/cdap/delta/app/DeltaPipelineId.java

```diff
@@ -42,4 +42,9 @@ public String getApp() {
   public long getGeneration() {
     return generation;
   }
+
+  @Override
+  public String toString() {
+    return namespace + "-" + app + "-" + generation;
```

Comment:
Think it would be better to move this whole class to the delta-api module and have getPipelineId() return this class instead of a stringified version.

Reply:
thanks for the suggestion:) Done :)